ENGAGEHF/Managers/VideoManager/VideoManager.swift

```diff
@@ -43,6 +45,12 @@ final class VideoManager: Module, EnvironmentAccessible, DefaultInitializable {
                     }
 
                     if event.newEnrolledAccountDetails != nil {
+                        // Refresh Firebase token to ensure proper permissions for accessing educational videos.
+                        do {
+                            try await Auth.auth().getToken(forcingRefresh: true)
```

Comment:
This fixes the specific issue, yes, but it isn't the most elegant nor complete solution to the problem in general. I'm fine with leaving it as is, but would like to give some context for discussion anyways:

- Right after the enrollUser call, we already get the new token (including the forceRefresh), but we do not wait for it before sending out the event that is handled here. This is also not easily possible, but we may want to remove that call, since this part makes it obsolete / redundant.

- This token refresh is technically not only related to the education videos, but would also cause the same issue for example on the Heart Health page. Can you please check, if the snapshot listeners silently fail there, too? It just isn't apparent there, since new accounts usually don't have vitals data and the issue is only visible right after the enrollUser call.

Technically, it would be best, if we find a solution where this token refresh occurs on a global scale and everyone gets notified about the new account after the token has been refreshed.

Reply:
A solution that is very easy to implement, but also a bit wasteful is just having that token refresh in every single of these listeners separately and then we just fetch the token like 5 times in parallel. Optimal would be to share the task and just awaiting it multiple times.

Reply:
I just talked with @PSchmiedmayer and one possible relatively simple solution would be to use a Lock (potentially exposed from the InvitationCodeModule or a separate token refresh) that all the listeners can check for before continuing with creating the snapshot listeners

i.e. after enrollUser (or maybe even right before it), we lock the lock and then each of the listeners (i.e. VideoManager and VitalsManager and MedicationsManager) will wait for that lock to be unlocked before creating their snapshot listeners.

Reply:
@nriedman Can you verify that the latest fixes on the main branch actually resolve the videos issue we were having? Would be amazing if you can iterate in this here and we can merge the PR as soon as possible 🚀

Reply:
I went ahead and tested out signing up with a new invitation code and immediately checking the Video Section, and the videos appeared every time 👍.

I'll go ahead and add that as a last step to the test that creates a new account using an invitation code.